crates/brightstaff/src/handlers/chat_completions.rs

```diff
@@ -33,54 +35,42 @@ pub async fn chat(
     let chat_request_bytes = request.collect().await?.to_bytes();
 
     debug!("Received request body (raw utf8): {}", String::from_utf8_lossy(&chat_request_bytes));
-
-    let chat_request_parsed = serde_json::from_slice::<serde_json::Value>(&chat_request_bytes)
-        .inspect_err(|err| {
-            warn!(
-                "Failed to parse request body as JSON: err: {}, str: {}",
-                err,
-                String::from_utf8_lossy(&chat_request_bytes)
-            )
-        })
-        .unwrap_or_else(|_| {
-            warn!(
-                "Failed to parse request body as JSON: {}",
-                String::from_utf8_lossy(&chat_request_bytes)
-            );
-            serde_json::Value::Null
-        });
-
-    if chat_request_parsed == serde_json::Value::Null {
-        warn!("Request body is not valid JSON");
-        let err_msg = "Request body is not valid JSON".to_string();
-        let mut bad_request = Response::new(full(err_msg));
-        *bad_request.status_mut() = StatusCode::BAD_REQUEST;
-        return Ok(bad_request);
-    }
-
-    let chat_completion_request: ChatCompletionsRequest =
-        serde_json::from_value(chat_request_parsed.clone()).unwrap();
-
-    // remove metadata from the request
-    let mut chat_request_user_preferences_removed = chat_request_parsed;
-    if let Some(metadata) = chat_request_user_preferences_removed.get_mut("metadata") {
-        if let Some(m) = metadata.as_object_mut() {
-            m.remove("archgw_preference_config");
-            debug!("Removed archgw_preference_config from metadata");
+    let provider_request = match ProviderRequestType::try_from((&chat_request_bytes[..], &SupportedAPIs::from_endpoint(request_path.as_str()).unwrap())) {
+        Ok(request) => request,
+        Err(err) => {
+            warn!("Failed to parse request as ProviderRequestType: {}", err);
+            let err_msg = format!("Failed to parse request: {}", err);
+            let mut bad_request = Response::new(full(err_msg));
+            *bad_request.status_mut() = StatusCode::BAD_REQUEST;
+            return Ok(bad_request);
         }
+    };
+
+    // Convert to ChatCompletionsRequest regardless of input type
+    let chat_completions_request_for_arch_router: ChatCompletionsRequest =
+        match ProviderRequestType::try_from((provider_request, &SupportedAPIs::OpenAIChatCompletions(hermesllm::apis::OpenAIApi::ChatCompletions))) {
```

Comment:
With this change you're effectively removing support for anthropic. Because now if someone sends an anthropic request you will fail right here.